lib/prepare-sundials.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+SUNDIALS_VERSION="4.0.1"
+
+SUNDIALS_TAR=sundials-${SUNDIALS_VERSION}.tar.gz
+
+tar xvzf $SUNDIALS_TAR
+mv sundials-${SUNDIALS_VERSION} sundials_${SUNDIALS_VERSION}
+
+cd sundials_${SUNDIALS_VERSION}
+
+rm -rf INSTALL_GUIDE.pdf config/ doc/ examples/ test/ */cvode */ida */arkode */kinsol
+find . -name CMakeLists.txt -exec rm {} \;
+
+find src -name "*.c" -type f -exec sed -E -i _orig  's#[^sf]printf\(#STAN_SUNDIALS_PRINTF(#'g {} \;
```

Comment:
I would prefer if we would actually commit to git the source directly from Sundials at this point prior to changing things. Even if it's just one commit followed by another, I think it'll help us back out changes if we mess up the process. What do you think?

Reply:
> Hmm... I would not do that as it complicates things. Moreover, git is super clever with these version changes and only tracks actual changes. Having those vanilla sundials -> adapted sundials commits in between bloats this.

I have the opposite reaction when I know this. 

I'm looking for easier review and maintenance. The overall result will be the same. 

If we have one commit that's the pruned Sundials and the next that's the changes to Sundials, I find using the GitHub interface it's easy to tell that the second commit actually does what we intend. And if we make a mistake, it's easy to back out that one commit. If the two actions aren't separated in git, then we can't tell and it's more work in verifying that this is correct. 

In fact, this would have helped with this current PR. Try thinking about the person reviewing. If I told you that I've updated a library and changed things, how would you verify it? My first thought was to download and do a diff, but I can't compare that to the final. If there's a better way to validate the PR, let me know. If not, let's add the additional git commit, which, as you say, git is clever and won't actually impact the user (but will positively impact the reviewer).